app/src/main/java/com/example/android/codelabs/navigation/mvp/presenter/MainActivityPresenter.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.android.codelabs.navigation.mvp.presenter
+
+import android.view.Menu
+import android.view.MenuItem
+import androidx.appcompat.app.AppCompatActivity
+import com.example.android.codelabs.navigation.mvp.contracts.MainActivityContracts
+
+class MainActivityPresenter(private val view: MainActivityContracts.View) : MainActivityContracts.Presenter{
+    override fun init() {
+        view.init()
+        view.setupActionBar()
+        view.setupBottomNavMenu()
+        view.setupNavigationMenu()
+        view.destinationChangedListener()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
```

Comment:
Here has the same case about the return...it's a single line..so you can use :Boolean  = ...

Reply:
done